### PR TITLE
feat(SNS): add topic_arn to created secrets

### LINKED ISSUE #1577: SNS Provider - Topic: feed the topicARN into secret
### What problem are you facing?

I'm implementing crossplane with AWS providers. 

Onto the creation of SNS Topics, I'm pretty happy because it creates the resources as I want and can fine tune them.

I also try to automate the injection of generated values into pods so that I don't have to go trough the AWS console and retrieve data to inject. 
It works like a charm for SQS Queues, the secret generated contains the URL of the queue, but when it comes to SNS Topics, the secret is empty. 
So either I'm not reading the doc properly, or this feature does not exists.

### What could help solve your problem?

On SNS Topics, whenever we are asking for secret creation, the generated secret could contains the topicARN, like we have the URL for SQS Queues.

### PR DESCRIPTION
**Pull request by darkanakin41**
### Description of your changes

Fixes #1577: inject topic ARN in the secrets

I have:

- [X] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-1.17` labels to auto-backport this PR if necessary.

### How has this code been tested
It was build and deploy to a test cluster. The resource was properly created and and the secret was set properly aswell:

image

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by darkanakin41** [comment]
@jbw976 I've managed to build and test it and it worked: 
image

File: config/sns/config.go [hunk 1]
```diff
@@ -52,5 +52,15 @@ func Configure(p *config.Provider) { //nolint:gocyclo
 			}
 			return diff, nil
 		}
+		// Add the topicARN into the secret
+		r.Sensitive.AdditionalConnectionDetailsFn = func(attr map[string]any) (map[string][]byte, error) {
+			conn := map[string][]byte{}
+
+			if a, ok := attr["arn"].(string); ok {
+				conn["topic_arn"] = []byte(a)
```
Comment (turkenf) [inline config/sns/config.go:60]: Is there a particular reason why you named the arn field `topic_arn`? If not, I'd prefer to be consistent with the underlying provider.

```suggestion
				conn["arn"] = []byte(a)
```

**Review by turkenf — COMMENTED** [review]
Hi @darkanakin41,

Thank you for your first contribution 🙂 I left a comment for you to consider. Also, I see in your explanations that you have successfully tested it. It would be great if you could share a screenshot that the arn came to the secret properly.

**Comment by darkanakin41** [comment]
> Hi @darkanakin41,
> 
> Thank you for your first contribution 🙂 I left a comment for you to consider. Also, I see in your explanations that you have successfully tested it. It would be great if you could share a screenshot that the arn came to the secret properly.

Hey @turkenf 
The screenshot is in both the comment and the initial message (i've updated it this morning) but here it is again:
image
I've not displayed the value because it contains some work informations. But it you need to see the value, let me know :)

**Comment by turkenf** [comment]
/test-examples="examples/sns/v1beta1/topic.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/sns/v1beta1/topic.yaml"

**Review by turkenf — APPROVED** [review]
Thank you @darkanakin41, LGTM.

**Pull request merged**